benchmarks/run_benchmarks.py

```diff
@@ -10,9 +10,14 @@
 from collections import namedtuple
 import os
 import re
+import resource
 import shutil
 import subprocess
 import sys
+from typing import TYPE_CHECKING
```

Comment:
I noticed type annotations are not used in this file. I'm happy to remove them to conform with your repository's style, however, I do believe it is a good habit to use them. I was careful to write them in a way that compatibility with Python >= 3.7 should remain, and should support < 3.7 as well. Specifically:

- Importing `TYPE_CHECKING` and using the subsequent `if` will prevent types from loading/polluting the namespace during normal runtime of the application.
- Every annotation is quoted. This prevents annoying issues that crop up from unknown types or Python versions that don't support unquoted type annotations.
- The use of `List` is necessary for older versions of Python, as it was not always possible to use `list` for this.